cmd/root.go

```diff
@@ -5,37 +5,53 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
+	"path"
 	"strings"
 
+	"github.com/adrg/xdg"
 	"github.com/cycloidio/terracognita/log"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
 var (
-	hclOut                    io.Writer
-	stateOut                  io.Writer
-	closeOut                  []io.Closer
+	hclOut   io.Writer
+	stateOut io.Writer
+
+	closeOut = make([]io.Closer, 0, 0)
+
 	include, exclude, targets []string
 	logsOut                   io.Writer
 
 	// RootCmd it's the entry command for the cmd on terracognita
 	RootCmd = &cobra.Command{
 		Use:   "terracognita",
 		Short: "Reads from Providers and generates a Terraform configuration",
-		Long:  "Reads from Providers and generates a Terraform configuration, all the flags can be used also with ENV (ex: --access-key == ACCESS_KEY)",
-		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		Long:  "Reads from Providers and generates a Terraform configuration, all the flags can be used also with ENV (ex: --aws-access-key == AWS_ACCESS_KEY)",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			err := os.MkdirAll(path.Dir(viper.GetString("log-file")), 0700)
+			if err != nil {
+				return err
+			}
+			logFile, err := os.OpenFile(viper.GetString("log-file"), os.O_APPEND|os.O_TRUNC|os.O_RDWR|os.O_CREATE, 0644)
+			if err != nil {
+				return err
+			}
+			closeOut = append(closeOut, logFile)
```

Comment:
no need to defer the call of `Close` in order to actually close the file handlers?

Reply:
We append all the `io.Closer` into this array so then when the CMD finishes can be closed all at once using `postRunEOutput` function.

Reply:
Ok great, did not see the `postRunEOutput` method :)